test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -423,19 +436,35 @@ var _ = ginkgo.Describe("end to end testing", func() {
 			originalArgs := util.ReadContainerArgsInDeployment(ctx, coreCluster, "wds1-system", "kubestellar-controller-manager", "manager")
 			originalArgsBytes, err := json.Marshal(originalArgs)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			originalArgsPatch := []byte(fmt.Sprintf(`{"spec":{"template":{"spec":{"containers":[{"args":%s,"name":"manager"}]}}}}`, string(originalArgsBytes)))
 			changedArgs := append(originalArgs, "--controllers=binding")
 			changedArgsBytes, err := json.Marshal(changedArgs)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			var scaledDown, semiCrashed bool
+			ginkgo.DeferCleanup(func(ctx context.Context) {
+				if semiCrashed {
+					_, err = coreCluster.AppsV1().Deployments("wds1-system").Patch(ctx, "kubestellar-controller-manager", types.StrategicMergePatchType, originalArgsPatch, metav1.PatchOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+				if scaledDown {
+					err = util.ScaleDeployment(ctx, coreCluster, "wds1-system", "kubestellar-controller-manager", 1)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			})
 
+			scaledDown = true
```

Comment:
Looks like `scaledDown` is
- set to `true` always _before_ calling `ScaleDeployment` _without_ error checking, and
- set to `false` always _after_ calling `ScaleDeployment` _with_ error checking.
Curious about why there are such differences between `true` and `false`.

Reply:
There is no way to atomically set it true together with the actual change being made in etcd.

It is safe to be true even when the object is not scaled down.